envparse.py

```diff
@@ -1,197 +1,217 @@
-# deal in the Software without restriction, including without limitation the
-# rights to use, copy, modify, merge, publish, distribute, sublicense, and/or
-# sell copies of the Software, and to permit persons to whom the Software is
-# furnished to do so, subject to the following conditions:
-#
-# The above copyright notice and this permission notice shall be included in
-# all copies or substantial portions of the Software.
-#
-# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
-# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
-# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
-# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
-# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
-# FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS
-# IN THE SOFTWARE.
-
+"""
+envparse is a simple utility to parse environment variables.
+"""
+from __future__ import unicode_literals
+import inspect
+import json as pyjson
+import logging
 import os
-import types
-import unittest
+import re
+import shlex
 import warnings
+try:
+    import urllib.parse as urlparse
+except ImportError:
+    # Python 2
+    import urlparse
 
 
-NOTSET = object()
+__version__ = '0.2.0'
 
 
-def env(var, cast=None, default=NOTSET):
-    """Return value for given environment variable.
+logger = logging.getLogger(__file__)
 
-    :param cast: Type to cast return value as.
-    :param default: If var not present in environ, return this instead.
 
-    :returns: Value from environment or default (if set)
-    """
-    try:
-        value = os.environ[var]
-    except KeyError:
-        if default is NOTSET:
-            raise
-
-        value = default
-
-    # Resolve any proxied values
-    if hasattr(value, 'startswith'):
-        if value.startswith('$'):
-            warnings.warn("$foo style proxied values will be removed in 1.0",
-                          DeprecationWarning, stacklevel=2)
-            # Old-style: For backwards compat. Won't work if passed through
-            # the shell without escaping
-            value = value.lstrip('$')
-            value = env(value, cast=cast, default=default)
-        if value.startswith('{') and value.endswith('}'):
-            # New-style: Always works
-            value = value.lstrip('{').rstrip('}')
-            value = env(value, cast=cast, default=default)
-
-    # Don't cast if we're returning a default value
-    if value != default:
-        if cast is bool:
-            value = int(value) != 0
-        elif isinstance(cast, list):
-            value = map(cast[0], [x for x in value.split(',') if x])
-        elif cast:
-            value = cast(value)
+class ConfigurationError(Exception):
+    pass
+
+
+# Cannot rely on None since it may be desired as a return value.
+NOTSET = type(str('NoValue'), (object,), {})
 
-    return value
+
+def shortcut(cast):
```

Comment:
Would prefer `_shortcut` just to make clear this is an internal function. (Makes it easier to change the implementation details later).